src/ts/components/listselector.ts

```diff
@@ -108,7 +116,9 @@ export abstract class ListSelector<Config extends ListSelectorConfig> extends Co
       listItem.label = this.config.translator(listItem);
     }
 
-    this.removeItem(key); // Try to remove key first to get overwrite behavior and avoid duplicate keys
+    // Try to remove key first to get overwrite behavior and avoid duplicate keys
+    this.removeItem(key); // FIXME: This will trigger an ItemRemoved and an ItemAdded event which should not be the case
```

Comment:
I think this is a matter of taste. One could also argue that there should not be an `ItemAdded` event when an overwrite happens, because it's been there before (count stays the same, thus nothing is "added").

Reply:
Should we keep the FIXME or do we say that this isn't defined so the events are fine? Maybe change the FIXME to a normal comment to just point that out?

Reply:
If this really is purely a matter of taste and does not have any negative impact that should be fixed sooner or later, then change to a normal comment.

If it has some kind of major drawback then keep the fixme.